src/connectors/geo-search/connectGeoSearch.js

```diff
@@ -0,0 +1,333 @@
+ * @property {LatLng} southWest The bottom left corner of the map view.
+ */
+
+/**
+ * @typedef {Object} CustomGeoSearchWidgetOptions
+ * @property {boolean} [enableRefineOnMapMove=true] If true, refine will be triggered as you move the map.
+ * @property {boolean} [enableGeolocationWithIP=true] If true, the IP will be use for the geolocation. If the `position` option is provided this option will be ignored, since we already refine the results around the given position. See [the documentation](https://www.algolia.com/doc/api-reference/api-parameters/aroundLatLngViaIP) for more informations.
+ * @property {LatLng} [position] Position that will be use to search around. <br />
+ * See [the documentation](https://www.algolia.com/doc/api-reference/api-parameters/aroundLatLng) for more informations.
+ * @property {number} [radius] Maximum radius to search around the position (in meters). <br />
+ * See [the documentation](https://www.algolia.com/doc/api-reference/api-parameters/aroundRadius) for more informations.
+ * @property {number} [precision] Precision of geo search (in meters). <br />
+ * See [the documentation](https://www.algolia.com/doc/api-reference/api-parameters/aroundPrecision) for more informations.
+ */
+
+/**
+ * @typedef {Object} GeoSearchRenderingOptions
+ * @property {Object[]} items The matched hits from Algolia API.
+ * @property {function(Bounds)} refine Sets a bounding box to filter the results from the given map bounds.
+ * @property {function()} clearMapRefinement Reset the current bounding box refinement.
+ * @property {function(): boolean} isRefinedWithMap Return true if the current refinement is set with the map bounds.
+ * @property {function()} toggleRefineOnMapMove Toggle the fact that the user is able to refine on map move.
+ * @property {function(): boolean} isRefineOnMapMove Return true if the user is able to refine on map move.
+ * @property {function()} setMapMoveSinceLastRefine Set the fact that the map has moved since the last refinement, should be call on each map move. The call to the function triggers a new rendering only when the value change.
+ * @property {function(): boolean} hasMapMoveSinceLastRefine Return true if the map has move since the last refinement.
+ * @property {Object} widgetParams All original `CustomGeoSearchWidgetOptions` forwarded to the `renderFn`.
+ * @property {LatLng} [position] The current position of the search.
+ */
+
+/**
+ * **GeoSearch** connector provides the logic to build a widget that will display results on a map. It also provides a way to search for results based on their position. The connector exposes APIs in order to manage the search experience (search on map interaction or control the interaction for example).
+ *
+ * Note that the GeoSearch connector takes leverage of the [geosearch](https://www.algolia.com/doc/guides/searching/geo-search) capabilities of Algolia. Your hits **must** have a `_geoloc` attribute in order to be passed to the rendering function.
```

Comment:
For now, they will not be displayed no errors is thrown. We filter on the property to avoid the error. Note that the connector only works with a single position.

https://github.com/algolia/instantsearch.js/blob/912f5e543a576fad0fcb993c8467eebc0d6b50f1/src/connectors/geo-search/connectGeoSearch.js#L259

Reply:
That's actually a pretty cool behaviour and less scary than `your hits must have a _geoloc attribute`

Reply:
What do you mean by that? I should update the requirements?